Modules/socketmodule.c

```diff
@@ -7926,7 +7927,7 @@ PyInit__socket(void)
 #ifdef  IPPROTO_VRRP
     PyModule_AddIntMacro(m, IPPROTO_VRRP);
 #endif
-#ifdef  IPPROTO_SCTP
+#if defined(IPPROTO_SCTP) && !defined(__EMSCRIPTEN__)
```

Comment:
I found a better way to deal with the issue. I guess you did the same mistake as I and added the ``skipif`` call after the check for AIX? That doesn't work because ``@requireSocket("AF_INET", "SOCK_STREAM", "IPPROTO_SCTP")`` tries to create an ``IPPROTO_SCTP`` socket and decorators are executed in reverse order! 

This works:

```
--- a/Lib/test/test_socket.py
+++ b/Lib/test/test_socket.py
@@ -43,6 +43,7 @@
 
 VSOCKPORT = 1234
 AIX = platform.system() == "AIX"
+EMSCRIPTEN = sys.platform == "Emscripten"
 
 try:
     import _socket
...

 @requireAttrs(socket.socket, "sendmsg")
-@unittest.skipIf(AIX, "IPPROTO_SCTP: [Errno 62] Protocol not supported on AIX")
 @requireSocket("AF_INET", "SOCK_STREAM", "IPPROTO_SCTP")
+@unittest.skipIf(AIX, "IPPROTO_SCTP: [Errno 62] Protocol not supported on AIX")
+@unittest.skipIf(EMSCRIPTEN, "IPPROTO_SCTP: aborts on Emscripten")
 class SendmsgSCTPStreamTest(SendmsgStreamTests, SendrecvmsgSCTPStreamTestBase):
```

Reply:
I see you've opened https://github.com/python/cpython/pull/30538 so I'll remove this change.

Reply:
#30538 does not work ... it's still running the decorator and crashing node.js